app/jobs/search_keyword_job.rb

```diff
@@ -5,22 +5,36 @@ class SearchKeywordJob < ApplicationJob
 
   def perform(search_result_id)
     search_result = SearchResult.find(search_result_id)
-    html_code = Search::SearchService.new(
-      keyword: search_result.keyword,
-      search_engine: search_result.search_engine
-    ).search
+    html_code = search(search_result.keyword, search_result.search_engine)
 
-    return mark_as_failed(search_result) unless html_code
+    result = parse(search_result.search_engine, html_code)
 
-    mark_as_completed(search_result, html_code)
-  rescue IcRailsHueyToby::Errors::SearchServiceError
+    mark_as_completed(search_result, html_code, result)
+  rescue IcRailsHueyToby::Errors::SearchServiceError, IcRailsHueyToby::Errors::ParseServiceError
     mark_as_failed(search_result)
   end
 
   private
 
-  def mark_as_completed(search_result, html_code)
-    search_result.update html_code: html_code, status: :completed
+  def search(keyword, search_engine)
+    Search::SearchService.new(
+      keyword: keyword,
+      search_engine: search_engine
+    ).search
+  end
+
+  def parse(search_engine, html_code)
+    Parse::ParseService.new(
+      search_engine: search_engine,
+      html_code: html_code
+    ).parse
+  end
+
+  def mark_as_completed(search_result, html_code, result)
+    search_result.update html_code: html_code, adwords_top_urls: result[:adwords_top_urls],
+                         adwords_top_count: result[:adwords_top_count], adwords_total_count: result[:adwords_total_count],
+                         non_adwords_urls: result[:non_adwords_urls], non_adwords_count: result[:non_adwords_count],
+                         total_links_count: result[:total_links_count], status: :completed
   end
```

Comment:
As you are listing all the keys of `result`, should we just pass the whole `result` instead?
```suggestion
  def mark_as_completed(search_result, html_code, result)
    search_result.update(result.merge(html_code: html_code, status: :completed))
  end
```